core/src/test/java/org/axonframework/XStreamEnumSetConverter.java

```diff
@@ -0,0 +1,54 @@
+package org.axonframework;
+
+import com.thoughtworks.xstream.converters.Converter;
+import com.thoughtworks.xstream.converters.MarshallingContext;
+import com.thoughtworks.xstream.converters.UnmarshallingContext;
+import com.thoughtworks.xstream.io.HierarchicalStreamReader;
+import com.thoughtworks.xstream.io.HierarchicalStreamWriter;
+
+import java.util.EnumSet;
+
+/**
+ * This converter would have to be registered for each enum that is serialized by Axon.
+ * Example:
+ * <pre>{@code
+ *  xStream.registerConverter(
+ *    new XStreamEnumSetConverter(TestEventWithEnumSet.SomeEnum.class)
+ *  );
+ * }</pre>
+ */
+public class XStreamEnumSetConverter implements Converter {
```

Comment:
We probably gonna need to register multiple converters like this one in CIM. We need to check how many of those CIM needs, also based on the pre-existing data in our customer's databases. Make sure this information is included in the readme.